src/ValueObjects/Money.php

```diff
@@ -0,0 +1,104 @@
+<?php
+namespace Give\ValueObjects;
+
+/**
+ * Class Money
+ * @package Give\ValueObjects
+ *
+ * @since 2.9.0
+ */
+class Money {
+	/**
+	 * @var int|string
+	 */
+	private $amount;
+
+	/**
+	 * @var int
+	 */
+	private $minorAmount;
+
+	/**
+	 * @var string
+	 */
+	private $currency;
+
+	/**
+	 * @var array
+	 */
+	private $currencyConfig;
+
+	/**
+	 * Return Money class object.
+	 *
+	 * @since 2.9.0
+	 *
+	 * @param int|string $amount Amount value without currency formatting
+	 * @param string $currency
+	 *
+	 * @return Money
+	 */
+	public static function of( $amount, $currency ) {
+		$object = new static();
+
+		$object->currencyConfig = give_get_currency_formatting_settings( $currency );
```

Comment:
Since this doesn't change, this can instead be a static method, `Money::currencyInfo()` so that each instance doesn't duplicate this array.

Reply:
fixed: https://github.com/impress-org/givewp/compare/e9d12bd15ddc...403590b3ded1 https://github.com/impress-org/givewp/compare/403590b3ded1...ae3b0cafc4f8